SoftLayer/CLI/virt/create.py

```diff
@@ -155,196 +143,110 @@ def _parse_create_args(client, args):
+@click.option('--image', help="Image ID. See: 'slcli image list' for reference")
+@click.option('--boot-mode', type=click.STRING,
+              help="Specify the mode to boot the OS in. Supported modes are HVM and PV.")
+@click.option('--billing', type=click.Choice(['hourly', 'monthly']), default='hourly', show_default=True,
               help="Billing rate")
-@click.option('--dedicated/--public',
-              is_flag=True,
-              help="Create a Dedicated Virtual Server")
-@click.option('--host-id',
-              type=click.INT,
-              help="Host Id to provision a Dedicated Host Virtual Server onto")
-@click.option('--san',
-              is_flag=True,
-              help="Use SAN storage instead of local disk.")
-@click.option('--test',
-              is_flag=True,
-              help="Do not actually create the virtual server")
-@click.option('--export',
-              type=click.Path(writable=True, resolve_path=True),
+@click.option('--dedicated/--public', is_flag=True, help="Create a Dedicated Virtual Server")
+@click.option('--host-id', type=click.INT, help="Host Id to provision a Dedicated Host Virtual Server onto")
+@click.option('--san', is_flag=True, help="Use SAN storage instead of local disk.")
+@click.option('--test', is_flag=True, help="Do not actually create the virtual server")
+@click.option('--export', type=click.Path(writable=True, resolve_path=True),
               help="Exports options to a template file")
 @click.option('--postinstall', '-i', help="Post-install script to download")
-@helpers.multi_option('--key', '-k',
-                      help="SSH keys to add to the root user")
+@helpers.multi_option('--key', '-k', help="SSH keys to add to the root user")
 @helpers.multi_option('--disk', help="Disk sizes")
-@click.option('--private',
-              is_flag=True,
+@click.option('--private', is_flag=True,
               help="Forces the VS to only have access the private network")
-@click.option('--like',
-              is_eager=True,
-              callback=_update_with_like_args,
+@click.option('--like', is_eager=True, callback=_update_with_like_args,
               help="Use the configuration from an existing VS")
 @click.option('--network', '-n', help="Network port speed in Mbps")
 @helpers.multi_option('--tag', '-g', help="Tags to add to the instance")
-@click.option('--template', '-t',
-              is_eager=True,
-              callback=template.TemplateCallback(list_args=['disk',
-                                                            'key',
-                                                            'tag']),
+@click.option('--template', '-t', is_eager=True,
+              callback=template.TemplateCallback(list_args=['disk', 'key', 'tag']),
               help="A template file that defaults the command-line options",
               type=click.Path(exists=True, readable=True, resolve_path=True))
 @click.option('--userdata', '-u', help="User defined metadata string")
```

Comment:
It seems the `--userData` is broken, I sent the following call and the SoftLayer_Virtual_Guest::getUserData returns an empty value, it doesn't happens with the current version.

```slcli vs create --hostname ajcb01-testslxi --domain softlayer.com --datacenter dal09 --ipv6 --os UBUNTU_LATEST -f BL2_1X2X100 --billing hourly -i http://local.test/image.txt --disk={100,25} -n 1000 --private --userdata="name:asdkjh, ajksdhkjhk" --public-security-group={1064939,128317,1085497} --private-security-group={1085497,648345}```

Reply:
Thanks, I'll double check that. Likely just needs to be in a slightly different format now that im using placeOrder.

Reply:
userData is working. I made a little more tests and it seems all is working so I think it can be merged.